AwaitExpception.Test/Tests.cs

```diff
@@ -34,5 +35,38 @@ public async Task TestFreeFallExceptionThrow()
             // make sure we have waited long enough
             Assert.True(thrower.AwaitCompleted);
         }
+
+        // This test fails the process with exception, not yet found a way to run it normally as a test.
+        // Comment that out if need to run other tests
+        [Test]
+        public async Task TestExceptionThrowInAsyncVoid()
+        {
+            var thrower = new ExceptionThrower();
+            AppDomain.CurrentDomain.UnhandledException += (object sender, UnhandledExceptionEventArgs e) => 
+            {
+                // Unfortunately I haven't yet found the way to not fail here. 
+                // Looks like the exception is caught but the process fails despite of that fact
+                Console.WriteLine("The exception is actually caught globally");

```

Comment:
Shouldn't we add this code after we know how to deal with it correctly? Right now it is just dead.

Reply:
No this is the key code here. But it doesn't work as expected for some reason. This is a generic mechanism for catching uncaught exceptions. So we know we expect an uncaught exception, but this method doesn't handle the exception it catches. The exception just propagates further for some reason. The console log here is important that in the console you will see a log that the exception was actually caught with this mechanism before it crashes the app.

Reply:
Ah, looks like ```UnhandledException``` is not intended to recover from the exception it just notifies that your application will fail in a moment so you for example can log something. And we are logging that. So looks like there is merely something we can do to prevent the crash. Looks like that is an example of a 100% crash if exception is raised.